### PR TITLE
🐟 Use ApiRx

### PR DESCRIPTION
**Pull request by jodator**

### CODE REVIEW

**Comment by krzysztof-jelski** [comment]
This looks definitely better IMO.

File: packages/ui/src/modals/TransferModal/SignTransferModal.tsx [hunk 1]
```diff
@@ -67,14 +60,20 @@ export function SignTransferModal({ onClose, from, amount, to }: Props) {
       onClose()
     }
 
-    if (keyringPair.meta.isInjected) {
-      const { signer } = await web3FromAddress(from.address)
+    let subscription: Subscription
 
-      await submittableExtrinsic.signAndSend(from.address, { signer: signer }, statusCallback).catch(console.error)
+    if (keyringPair.meta.isInjected) {
+      web3FromAddress(from.address).then(({ signer }) => {
```
Comment (krzysztof-jelski) [inline packages/ui/src/modals/TransferModal/SignTransferModal.tsx:66]: I guess it's still ok to use `await` here.

File: packages/ui/src/providers/api/context.tsx [hunk 2]
```diff
@@ -3,6 +3,5 @@ import { UseApi } from './provider'
 
 export const ApiContext = createContext<UseApi>({
   isConnected: false,
-  state: 'CONNECTING',
```
Comment (krzysztof-jelski) [inline packages/ui/src/providers/api/context.tsx:6]: Can't we detect connecting state with RxApi?

File: packages/ui/src/hooks/useObservable.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export function useObservable<T>(observable: Observable<T> | undefined, deps: readonly any[]) {
+  const [data, setData] = useState<T | undefined>(undefined)
+
+  useEffect(() => {
+    const subscription = observable?.subscribe(setData)
+    return () => subscription && subscription.unsubscribe()
+  }, deps)
+
+  return [data]
```
Comment (krzysztof-jelski) [inline packages/ui/src/hooks/useObservable.ts:12]: Why an array?

Comment (jodator) [reply packages/ui/src/hooks/useObservable.ts:12]: Hmmm, I've copied that part from the other project, but looks like I didn't give a second look at that. I don't see a reason actually.

File: packages/ui/src/providers/api/context.tsx (hunk 2 above)
Comment (jodator) [reply packages/ui/src/providers/api/context.tsx:6]: Probably everything between starting the connection and `isConnected` observable emits change is a connecting state. For now, it was not used so I've removed the `state` variable.

File: packages/ui/src/modals/TransferModal/SignTransferModal.tsx (hunk 1 above)
Comment (jodator) [reply packages/ui/src/modals/TransferModal/SignTransferModal.tsx:66]: I'm not sure if I can return Promise from the `useEffect()` hook callback (using the `async` keyword). I wanted to return the `subscription` callback so I've used the synchronous arrow funciton.

**Review by krzysztof-jelski — APPROVED** [review]

File: packages/ui/src/providers/api/context.tsx (hunk 2 above)
Comment (jodator) [reply packages/ui/src/providers/api/context.tsx:6]: I'd go with a holistic approach to connection status improvements: https://github.com/Joystream/pioneer/issues/76. For now, I'm OK with a current connection state handling.

Comment (krzysztof-jelski) [reply packages/ui/src/providers/api/context.tsx:6]: OK

**Pull request merged**